samples/sensor/sht3xd/prj_nrf.conf

```diff
@@ -0,0 +1,14 @@
+#
+# Copyright (c) 2018 Peter Bigot Consulting, LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_I2C=y
+CONFIG_HAS_DTS_I2C=y
+CONFIG_I2C_NRFX=y
+CONFIG_I2C_0=y
```

Comment:
I'm not sure if this is also true on nordic parts, but on nxp parts you don't have to set the three configs above. If you set CONFIG_I2C=y, then the soc configures the appropriate i2c driver and the board configures the appropriate i2c instance. @carlesc, do you guys do this too?

If this is the same on nordic parts, then you can eliminate `prj_nrf.conf` entirely.

Reply:
The default `prj.conf` results in errors about: 'drivers__i2c' was created without source files
I can and will get rid of `CONFIG_HAS_DTS_I2C` because that comes with the board, but the other two appear to be necessary.

Reply:
@MaureenHelm I think @Mierunski and @anangl should answer this question

Reply:
I've removed all the optional pieces.  If there are changes necessary to make this file unnecessary they're probably in board files, which I propose should be a separate PR.

Reply:
> The default `prj.conf` results in errors about: 'drivers__i2c' was created without source files

Ok, then it looks like the nordic parts don't handle this like I'd hoped. @Mierunski, @anangl you might want to consider doing this because it allows for fewer soc-specific .conf files in the sample and test applications.

https://github.com/zephyrproject-rtos/zephyr/blob/b5ae7da2697b6e38180856f014cfe6bbc652e4e6/soc/arm/nxp_kinetis/k6x/Kconfig.defconfig.mk64f12#L46-L51

https://github.com/zephyrproject-rtos/zephyr/blob/b5ae7da2697b6e38180856f014cfe6bbc652e4e6/boards/arm/frdm_k64f/Kconfig.defconfig#L73-L81